homeassistant/components/homeassistant/scene.py

```diff
@@ -143,12 +144,12 @@ def scenes_with_entity(hass: HomeAssistant, entity_id: str) -> list[str]:
     if DATA_PLATFORM not in hass.data:
         return []
 
-    platform = hass.data[DATA_PLATFORM]
+    platform: EntityPlatform = hass.data[DATA_PLATFORM]
 
     return [
         scene_entity.entity_id
         for scene_entity in platform.entities.values()
-        if entity_id in scene_entity.scene_config.states
+        if entity_id in cast(HomeAssistantScene, scene_entity).scene_config.states
```

Comment:
We could avoid the casting here.

```py
    scene_entities: list[HomeAssistantScene] = platform.entities.values()
    return [
        scene_entity.entity_id
        for scene_entity in scene_entities
        if entity_id in scene_entity.scene_config.states
    ]
```

Reply:
This suggestion is not working:
```
Incompatible types in assignment (expression has type "dict_values[str, Entity]", variable has type "List[HomeAssistantScene]")  [assignment]
```

Reply:
Oh I see, maybe change it to:

```py
    return [
        scene_entity.entity_id
        for scene_entity in cast(
            ValuesView[HomeAssistantScene], platform.entities.values()
        )
        if entity_id in scene_entity.scene_config.states
    ]
```

That way, we only call cast once.

Reply:
Done